### PR TITLE
Simplified creating schema for parquet files

### PR DESCRIPTION
**Pull request by norberttech**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    <li>static factories to Flat and Nested columns in parquet schema</li>
    <li>Decimal column support to parquet</li>
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    <li>Calculation of max definitions/repetitions level to column</li>
    <li>Python parquet file generators compression from gzip to snappy</li>
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

### CODE REVIEW

**Comment by norberttech** [comment]
@stloyd I merged it since it's not affecting anything critical in the ETL right now, but feel free to review it and leave comments, I still have plenty of iterations ahead of me so will adjust the code in the next PR.

File: src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema.php [hunk 1]
```diff
@@ -19,12 +18,20 @@ final class Schema
...
+    private function __construct(
         private readonly FlatColumn $schemaRoot,
         private readonly array $columns
     ) {
     }
 
+    public static function from(Column ...$columns) : self
+    {
+        return new self(
+            FlatColumn::boolean('schema'),
```
Comment (stloyd) [inline src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema.php:30]: Can we have phpdoc why this is set here? From the name, it's not clear that there is such a setting

**Review by stloyd — COMMENTED** [review]
.

File: src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/FlatColumn.php [hunk 2]
```diff
@@ -213,6 +273,12 @@ public function scale() : ?int
         return $this->scale;
     }
 
+    public function setParent(NestedColumn $parent) : void
+    {
+        $this->rootPath = $parent->flatPath();
```
Comment (stloyd) [inline src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/FlatColumn.php:278]: What if a parent was already set?

File: src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/NestedColumn.php [hunk 3]
```diff
@@ -17,11 +17,57 @@ final class NestedColumn implements Column
...
         private readonly FlatColumn $root,
         array $columns,
-        private readonly int $maxDefinitionsLevel,
-        private readonly int $maxRepetitionsLevel,
         private ?Column $parent = null
     ) {
         $this->children = $columns;
+
+        foreach ($columns as $column) {
+            if ($column->parent() === null) {
```
Comment (stloyd) [inline src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/NestedColumn.php:25]: Is it possible that the column has a different parent?

File: src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema.php (hunk 1 above)
Comment (norberttech) [reply src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema.php:30]: it was just the first step before #622

File: src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/FlatColumn.php (hunk 2 above)
Comment (norberttech) [reply src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/FlatColumn.php:278]: irrelevant anymore, creating a rootPath is now dynamic

File: src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/NestedColumn.php (hunk 3 above)
Comment (norberttech) [reply src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/NestedColumn.php:25]: No, but it was possible that the parent was already set due to a misalignment between rebuilding schema from thrift and code directly.

**Pull request merged**